postgres/init.sls

```diff
@@ -126,6 +147,7 @@ postgresql-db-{{ name }}:
     - lc_ctype: {{ db.get('lc_ctype', 'en_US.UTF8') }}
     - lc_collate: {{ db.get('lc_collate', 'en_US.UTF8') }}
     - template: {{ db.get('template', 'template0') }}
+    - tablespace: {{ db.get('tablespace', 'pg_default') }}
```

Comment:
> it is an error to specify a database with a tablespace that doesn't exist, therefore there needs to be a dependency in that direction

It **is** an error, agreed. But the way I see it, specifying a dependency on a tablespace that does not exist is no different an error than specifying a package name in your pillar that does not exist (like `package: psotgresql-sevrer`).

I also think it's OK to depend on things you are not managing with Salt (yet) or that out of your control, but can be used as a dependency for what you do. And if they do not exist, it's  OK for the state to *fail*, because in the end, you *must* know your infrastructure to automate it and that unmet dependency *shows* you there's an error in the current infrastructure (and/or what you think it should be) and does not proceed any further.

What you're doing in PR #109 is to make *mandatory* that the tablespace is managed with Salt to be usable as a dependency in the database state. And I think that might *lead* to errors instead of just showing them:

Consider the following hypothetical case (I understand it's a corner case, but possible):

 * The package installer for the app **my_db_app** creates a tablespace **mydbapp_tbsp**, where it stores data, with tablespace owner **mydbapp**. In salt, I just add a package state to install that package.
* I want, using this formula, to create a database in that tablespace for some statistics, so I add a database state with `name: statistics` and `tablespace: my_db_app`. Because *I know* what I'm installing and configuring, I know the previously added package space will add the tablespace, and all will go OK.
 * One day, new version of package **my_db_app** changes the owner of the tablespace to **someotherowner**. 

This PR will still work as expected, because the states are ordered and depedendencies are set but not enforced. 

Now, consider the same example under PR #109: you're making *mandatory* to add a tablespace state to create **mydbapp_tbsp** with owner **mydbapp**. When the package **my_db_app** is updated and it changes the owner of that tablespace to **someotherowner**, your *forced* tablespace state will *revert* it back to **mydbapp**. And break things.

Again, is a corner case. But entirely possible, as described or with variants. In those cases, I prefer to offer the possibility of adding the dependency than enforcing it.